File: src/enemy/commands/attack.py
import discord

from .command import Command
from enemy import db

from enemy.systems import combat
from enemy.env import config


class Attack(Command):
    def process(message: discord.Message):
        attacker = db.get_user(message.author.id)
        attackee = db.get_user(config.enemy_id)

        attacker, attackee = combat.attack(attacker, attackee)

        db.set_user(attacker)
        db.set_user(attackee)

File: packages/discord/src/commands/attack.ts
import { Message } from "discord.js";

import { getUser, setUser, createUser, getUserInventory } from "../api";
import { variables } from "../env";

import { attack } from "../systems/combat";
import { sendMessage } from "../helpers/message";

export async function attackCommand(message: Message) {
  let attacker = await getUser(message.author.id);
  let attackerInventory = await getUserInventory(attacker);

  const attackerState = 

  if (!attacker) {
    attacker = await createUser({ discordId: message.author.id });
  }

  let attackee = await getUser(variables.enemyId);

  const { newAttackerState, newAttackeeState } = attack(attacker, attackee);

  setUser(newAttackerState);
  setUser(newAttackeeState);

  sendMessage(message, "ouch", true);
}
